### PR TITLE
feat(sheets-drawing-ui): support sheet cell image

### PR DESCRIPTION
**Pull request by weird94**
close #xxx

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

File: packages/core/src/sheets/typedef.ts [hunk 1]
```diff
@@ -290,6 +290,8 @@ export interface ICellDataForSheetInterceptor extends ICellData {
         downOffset?: number;
         isSkip?: boolean;
     };
+
+    images?: Record<string, HTMLImageElement>;
```
Comment (wzhudev) [inline packages/core/src/sheets/typedef.ts:294]: 这个方法不是很好，违反了开闭原则。看看有没有其他自定义渲染的更佳方式？ cc @lumixraku

File: packages/sheets-drawing-ui/src/controllers/sheet-celll-image-hover.controller.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    constructor(
+        @Inject(HoverManagerService) private _hoverManagerService: HoverManagerService,
+        @Inject(IRenderManagerService) private _renderManagerService: IRenderManagerService,
+        @Inject(SheetsSelectionsService) private _selectionsService: SheetsSelectionsService,
+        @Inject(SheetCellCacheManagerService) private _sheetCellCacheManagerService: SheetCellCacheManagerService,
+        @Inject(DrawingRenderService) private _drawingRenderService: DrawingRenderService
+    ) {
+        super();
+        // this._initHover();
```
Comment (wzhudev) [inline packages/sheets-drawing-ui/src/controllers/sheet-celll-image-hover.controller.ts:36]: 这里的代码被注释掉了，这是预期的吗？

File: packages/sheets-drawing-ui/src/services/sheet-cell-cache-manager.service.ts [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+        return makeImageRecord(this._getByPosition(row, col));
+    }
+
+    getByKey(row: number, col: number, key: string) {
+        return this._poistionMap.getValue(row, col)?.get(key)?.image;
+    }
+}
+
+export class SheetCellCacheManagerService extends Disposable {
+    private _imageMaps = new Map<string, Map<string, ImageCacheMap>>();
```
Comment (wzhudev) [inline packages/sheets-drawing-ui/src/services/sheet-cell-cache-manager.service.ts:97]: 单元格图片和浮动图片是否需要区别这么开呢？有一个功能叫做单元格图片转浮动图片，它们可能有比较紧密的关联。

**Comment by weird94** [comment]
> ## View Deployment
> 📑 Examples	📚 Storybook
> [🔗 Preview link](https://univer-2c3a6ayu3-univer.vercel.app/)	[🔗 Preview link](https://univer-design-pnxrgbmbx-univer.vercel.app/)

1

**Comment by weird94** [comment]
https://github.com/dream-num/univer/pull/4036#issuecomment-2468214407

File: packages/core/src/docs/data-model/text-x/apply-utils/update-apply.ts [hunk 4]
```diff
@@ -351,8 +351,7 @@ function updateCustomBlocks(
     currentIndex: number,
     coverType: UpdateDocsAttributeType
 ) {
-    const { customBlocks } = body;
-
+    const { customBlocks = [] } = body;
```
Comment (Jocs) [inline packages/core/src/docs/data-model/text-x/apply-utils/update-apply.ts:354]: 把空 customBlocks 处理成空数组会有问题吗？之前记得约定undefined是不处理，空数组是清空

Comment (weird94) [reply packages/core/src/docs/data-model/text-x/apply-utils/update-apply.ts:354]: 这是body不是updateBody

**Review by Jocs — APPROVED** [review]

**Pull request merged**